reconcile/openshift_resources_base.py

```diff
@@ -1198,57 +1196,22 @@ def _early_exit_fetch_resource(spec: Sequence, settings: Mapping) -> dict[str, s
 @contextmanager
 def early_exit_monkey_patch() -> Generator:
     """Avoid looking outside of app-interface on early-exit pr-check."""
-    orig_lookup_secret = lookup_secret
-    orig_lookup_github_file_content = lookup_github_file_content
-    orig_url_makes_sense = url_makes_sense
-    orig_check_alertmanager_config = check_alertmanager_config
-    orig_lookup_s3_object = lookup_s3_object
-
-    try:
-        _early_exit_monkey_patch_assign(
-            lambda path,
-            key,
-            version=None,
-            tvars=None,
-            allow_not_found=False,
-            settings=None,
-            secret_reader=None: f"vault({path}, {key}, {version}, {allow_not_found})",
-            lambda repo,
-            path,
-            ref,
-            tvars=None,
-            settings=None,
-            secret_reader=None: f"github({repo}, {path}, {ref})",
-            lambda url: False,
-            lambda data, path, alertmanager_config_key, decode_base64=False: True,
-            lambda account_name,
-            bucket_name,
-            path,
-            region_name=None: f"lookup_s3_object({account_name}, {bucket_name}, {path}, {region_name})",
-        )
-        yield
-    finally:
-        _early_exit_monkey_patch_assign(
-            orig_lookup_secret,
-            orig_lookup_github_file_content,
-            orig_url_makes_sense,
-            orig_check_alertmanager_config,
-            orig_lookup_s3_object,
-        )
-
-
-def _early_exit_monkey_patch_assign(
-    lookup_secret: Callable,
-    lookup_github_file_content: Callable,
-    url_makes_sense: Callable,
-    check_alertmanager_config: Callable,
-    lookup_s3_object: Callable,
-) -> None:
-    sys.modules[__name__].lookup_secret = lookup_secret  # type: ignore
-    sys.modules[__name__].lookup_github_file_content = lookup_github_file_content  # type: ignore
-    sys.modules[__name__].url_makes_sense = url_makes_sense  # type: ignore
-    sys.modules[__name__].check_alertmanager_config = check_alertmanager_config  # type: ignore
-    sys.modules[__name__].lookup_s3_object = lookup_s3_object  # type: ignore
+    with patch.multiple(
+        jinja2_utils,
+        lookup_secret=DEFAULT,
+        lookup_github_file_content=DEFAULT,
+        url_makes_sense=DEFAULT,
+        lookup_s3_object=DEFAULT,
+    ) as mocks:
+        mocks["lookup_secret"].return_value = "secret"
+        mocks["lookup_github_file_content"].return_value = "file-content"
+        mocks["url_makes_sense"].return_value = False
+        mocks["lookup_s3_object"].return_value = "s3-object"
```

Comment:
previously we return value based on called args, so even not do real fetch, still return some identifier back, do we need that?

Reply:
I think we don't need that.

Reply:
I think it's interesting to keep a return value that relates to arguments. If a template is updated with a parameter change of one of these calls (eg a vault secret version), we want the change to be detected and no early exit. WDYT?

Reply:
Good point!

Reply:
Changed in [30dd5a2](https://github.com/app-sre/qontract-reconcile/pull/4386/commits/30dd5a214b04cebd78d10f7081e4ad3f9f474afc)

Reply:
so in the end, what do we fix in early exit?
I agree this refactor is nicer than teh initial code, but I don't think it fixes anything, right?

Reply:
It just fixes the mocking of the template functions (`vault,` `github,`...), and this should now speed up the calculation of the early exit state.

Reply:
I don't see how we fix it. It used to return a f-string, and it still does.

But fine, I like the use of mocks much more ;-)

Reply:
aaaaaah cool, thanks!